### DISCUSSION TITLE
How to add regular middleware to specific routes?

### DISCUSSION THREAD

**Discussion by samburgers** [RobinTail/express-zod-api#299] [Q&A] [answered]
Just wondering how best to compose any regular express middlewares such as this `checkJwt` middleware?

Looking to include this middleware to a few EZ endpoints, and figure there might be a way to do it without resorting to `app.get` routing.

Thanks!

https://auth0.com/docs/quickstart/backend/nodejs/01-authorization#protect-api-endpoints

```
app.get('/api/private', checkJwt, function(req, res) {
  res.json({
    message: 'Hello from a private endpoint! You need to be authenticated to see this.'
  });
});
```

**Comment by RobinTail** [comment, marked as answer]
Hello @samburgers ,

thank you for the very interesting question.
According to the documentation, `checkJwt` is the `Handler` created by `auth()` method, specified here:
https://github.com/auth0/node-oauth2-jwt-bearer/blob/0b65192/packages/express-oauth2-jwt-bearer/src/index.ts#L62

That Handler has the usual signature for express:
```typescript
(req: Request, res: Response, next: NextFunction) => Promise<void>
```

Request and Response are the entities available to `Middleware`, so you can create one the following way:
```typescript
const myJWTMiddleware = createMiddleware({
  input: z.object({}),
  middleware: async ({request, response}) => {
    const next = (e?: Error) => {
      if (e) {
        throw e;
      }
    };
    await checkJwt(request, response, next);
    return { auth: request.auth }; // returns become options for endpoint handler
  }
});
```

I did not try it in action, but it should be something similar.
I hope you'll find it helpful.
Please let me know.

You gave me an interesting idea. I think I will create a special method to facilitate this kind of middlewares.

**Reply by samburgers** [reply]
Thanks @RobinTail this works great! 

Such incredible work with this project :)
